### PR TITLE
Cross Posting Blog Posts - GraphQL - Diving Deep Series

### PR DESCRIPTION
**Pull request by tvvignesh**
Hi. Originally posted these blogs here: 

Part 1: https://medium.com/timecampus/graphql-diving-deep-c7c0abe608b2

Part 2: https://medium.com/timecampus/graphql-the-usecase-architecture-76e7e932d8de

Part 3: https://medium.com/timecampus/graphql-the-stack-1-99a8e4f820c2

Part 4: https://medium.com/timecampus/graphql-the-stack-2-fc13cfac5e90

@dotansimha had PMed me asking if I can cross post The GraphQL series I am working on in the Guild blog as well.

Hence doing the same with all branding removed.

**PS:** I will publish the rest of the GraphQL series as well as I write it. This is the place where I update when I first post the blog: https://github.com/graphql/graphql-wg/issues/540

### CODE REVIEW

**Comment by tvvignesh** [comment]
Will send a PR for the other 2 blogs as well now.

**Comment by tvvignesh** [comment]
@dotansimha The blogs I published so far on GraphQL has been added to this PR. Have tested it running locally. Hope this works.

**Comment by dotansimha** [comment]
This is awesome. Thank you @tvvignesh ! 
@Urigo can you please take a look and merge if ready?

**Comment by tvvignesh** [comment]
Updated this PR adding Part 4 as of today.

**Comment by Urigo** [comment]
Thank you so much @tvvignesh !
I'm reading through the posts and will give a proper review in the new couple of days!

File: pages/blog/graphql-deep-dive-1.mdx
Comment (Urigo) [inline pages/blog/graphql-deep-dive-1.mdx:16]: Would be great to add link to all the other parts of the series in all the articles

File: pages/blog/graphql-deep-dive-1.mdx
Comment (Urigo) [inline pages/blog/graphql-deep-dive-1.mdx:20]: I would add here that GraphQL is no longer in Facebook's control.  Mostly because I still see a lot of people referencing it and worried about previous licenses and stuff

Comment (Urigo) [inline pages/blog/graphql-deep-dive-1.mdx:20]: Instead of awesome-graphql, which is not ordered and hard to look at, I would reference the new https://graphql.org/code/

File: pages/blog/graphql-deep-dive-1.mdx
Comment (Urigo) [inline pages/blog/graphql-deep-dive-1.mdx:134]: I would also mention here Relay and URQL, or maybe even just a simple react-query or graphql-request

File: pages/blog/graphql-deep-dive-1.mdx
Comment (Urigo) [inline pages/blog/graphql-deep-dive-1.mdx:144]: I would add here http://graphql-rules.com and personally I would be careful following principledgraphql.com as it is very opinionated towards a certain product....

File: pages/blog/graphql-deep-dive-1.mdx
Comment (Urigo) [inline pages/blog/graphql-deep-dive-1.mdx:244]: I would also add some points from the bottom of this page: https://the-guild.dev/about-us

File: pages/blog/graphql-deep-dive-1.mdx
Comment (Urigo) [inline pages/blog/graphql-deep-dive-1.mdx:255]: Could you add this one as well? https://product.voxmedia.com/2020/11/2/21494865/to-federate-or-stitch-a-graphql-gateway-revisited

File: pages/blog/graphql-deep-dive-1.mdx
Comment (Urigo) [inline pages/blog/graphql-deep-dive-1.mdx:275]: not sure if you want, but that might be a good reference: https://the-guild.dev/blog/migrating-from-rest

File: pages/blog/graphql-deep-dive-1.mdx
Comment (Urigo) [inline pages/blog/graphql-deep-dive-1.mdx:298]: A bit of self promotion but I think this talk covers a lot of basics around GraphQL Subscriptions - https://youtu.be/Wi7P39sF2nw

**Review by Urigo — APPROVED** [review]
This is just great!
I've made a couple of very small comments but you can feel free to ignore them completely, I approve without requiring any change.

Thank you so much for doing this, its so valuable and so missing in our current ecosystem

**Comment by Urigo** [comment]
also some things to consider about cross posting on Medium:

https://dev.to/seckelberny67/cross-post-from-medium-2mbc

https://dev.to/madza/is-it-worth-to-cross-post-on-medium-in-2020-3l38

File: pages/blog/graphql-deep-dive-1.mdx
Comment (tvvignesh) [reply pages/blog/graphql-deep-dive-1.mdx:16]: Sure. Will do it. Was planning to do it after writing all the blogs - will add links to the blogs added so far.

File: pages/blog/graphql-deep-dive-1.mdx
Comment (tvvignesh) [reply pages/blog/graphql-deep-dive-1.mdx:20]: Sure. Will emphasize more about the licensing and the foundation

Comment (tvvignesh) [reply pages/blog/graphql-deep-dive-1.mdx:20]: Sure. But I will add it as an additional link instead of removing `awesome-graphql` since I am not sure what projects are missing from either of the lists.

File: pages/blog/graphql-deep-dive-1.mdx
Comment (tvvignesh) [reply pages/blog/graphql-deep-dive-1.mdx:134]: Sure. Will add it.

File: pages/blog/graphql-deep-dive-1.mdx
Comment (tvvignesh) [reply pages/blog/graphql-deep-dive-1.mdx:144]: Sure. I will add the link as well. The main difference I see between `graphql-rules` and `principledgraphql` is that principledgraphql talks about the concepts in general and not the implementation itself, but I see `graphql-rules` talking about the implementation itself. So, I am not sure what is currently a replacement, but I will add both.

File: pages/blog/graphql-deep-dive-1.mdx
Comment (tvvignesh) [reply pages/blog/graphql-deep-dive-1.mdx:244]: Sure. Will do.

File: pages/blog/graphql-deep-dive-1.mdx
Comment (tvvignesh) [reply pages/blog/graphql-deep-dive-1.mdx:255]: Sure. Will add it, but I see the blog more as Schema stitching vs Federation rather than a case study - it may not belong in this section but will add it in another place.

File: pages/blog/graphql-deep-dive-1.mdx
Comment (tvvignesh) [reply pages/blog/graphql-deep-dive-1.mdx:275]: Sure. Will add the reference.

File: pages/blog/graphql-deep-dive-1.mdx
Comment (tvvignesh) [reply pages/blog/graphql-deep-dive-1.mdx:298]: Haha. Just watched the video. Nice talk for beginners. Will add it.

**Comment by tvvignesh** [comment]
> also some things to consider about cross posting on Medium:

Yup. Completely agree. I was actually intending to post on my own website blog to get some traction, but since I don't have any product anyways now, that didn't make sense now.

Then after thinking a while, I decided that I should not worry about SEO much now since my intention mainly should be to help people with information about OSS which they might not know. Hence I decided to publish on all blogging platforms wherever I can. I was thinking of cross-posting on `Dev,to` as well initially so that some more people may benefit from it.

To be honest, I have no alignment to Medium or any platform for that matter as long as people are able to discover and benefit from what I write. Let's see. I have not been liking Medium's paywall which restricts me from reading featured stories. But, since they also have a business to run, I get it.

File: pages/blog/graphql-deep-dive-1.mdx
Comment (tvvignesh) [reply pages/blog/graphql-deep-dive-1.mdx:255]: The main reason I feel that way is cause, it does not talk about how Vox uses Stitching, but just about stitching vs federation in general. While all the info is very useful, the context of the business is missing here.

**Comment by tvvignesh** [comment]
Thanks @Urigo for all the feedback - Will make the changes as necessary and update this PR within 2 days. Btw, are the rest of the blogs (Part 2, 3 and 4) good to go?

**Comment by Urigo** [comment]
Yes I read it all!

**Comment by tvvignesh** [comment]
@Urigo @dotansimha Have applied and pushed all the changes as suggested above. Thanks.

**Comment by Urigo** [comment]
@tvvignesh when will you want us to publish all of those and how?

Also what dates should we put on the posts?
The blog orders them by dates so maybe a day after the latest public post and all the rest will follow each other day by day?

After that we could just merge and publish them all, even today

**Comment by tvvignesh** [comment]
@Urigo I will leave that to you guys. Feel free to publish whenever you think is right to do it. Yup.

> The blog orders them by dates so maybe a day after the latest public post and all the rest will follow each other day by day?

Yup. Realized that. That's why I have put the dates on the blog posts one day after other to maintain ordering (but those were just done to maintain the order of the posts)

I have the next blog post coming up on Dec 5th which I will be sharing as well. Will send a PR for the next post by Dec 5th.

As you already know, if the readers have to come back to the blog periodically, a strict discipline/schedule with good quality content would be needed. So, maybe the future posts can happen weekly as I send a PR for that but I will leave the schedule completely to you.

Let me know if I have to update the posts with a certain date. Thanks.

**Comment by Urigo** [comment]
so we'll publish all of those today and from now on the rest will be published as you go?

I'm ok with that.
The other option would be to publish one today, and stretch them with time, but on the other hand, it's already out there so why delay...

**Comment by tvvignesh** [comment]
@Urigo Makes sense. Sure. So, I will update all the posts to have today's date (I guess I might have to add time as well else ordering will go for a toss)?

**Comment by Urigo** [comment]
The latest post was on the 30th
So maybe do 1, 2, 3 and 4 of December?
(even though 4 is tomorrow, I think it's fine :) )

**Comment by tvvignesh** [comment]
@Urigo Updated and pushed. Btw, tried adding timestamp but that was not respected. So, what you said would work.

**Comment by Urigo** [comment]
Thank you so much!!!

**Pull request merged**
